java/vector/src/main/java/org/apache/arrow/vector/types/Dictionary.java

```diff
@@ -18,23 +18,48 @@
  ******************************************************************************/
 package org.apache.arrow.vector.types;
 
-import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.FieldVector;
+
+import java.util.Objects;
 
 public class Dictionary {
 
-    private ValueVector dictionary;
-    private boolean ordered;
+  private Long id;
+  private FieldVector dictionary;
+  private boolean ordered;
+
+  public Dictionary(FieldVector dictionary) {
+    this(dictionary, null, false);
+  }
+
+  public Dictionary(FieldVector dictionary, Long id, boolean ordered) {
+    this.id = id;
+    this.dictionary = dictionary;
+    this.ordered = ordered;
+  }
+
+  public Long getId() { return id; }
+
+  public FieldVector getVector() {
+  return dictionary;
+  }
 
-    public Dictionary(ValueVector dictionary, boolean ordered) {
-        this.dictionary = dictionary;
-        this.ordered = ordered;
-    }
+  public boolean isOrdered() {
+  return ordered;
+  }
 
-    public ValueVector getDictionary() {
-        return dictionary;
-    }
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    Dictionary that = (Dictionary) o;
+    return id == that.id &&
+      ordered == that.ordered &&
+      Objects.equals(dictionary, that.dictionary);
```

Comment:
I needed to define equals to match with hashcode for collecting dictionaries during encoding. I'll revisit in light of defining dictionary ids up front.